File: MyToolTracker-backend/MyToolTrackerAPI/Models/Tool.cs
using System;
namespace MyToolTrackerAPI.Models
{
	public class Tool
	{
        public int Id { get; set; }
		public string Name { get; set; }
		public string Barcode { get; set; }
		public decimal Price { get; set; }
		public string Model { get; set; }
		public string Manufacturer { get; set; }
		public DateTime EntryDate { get; set; }

        // Foreign keys
        public int CategoryId { get; set; }

        // Relationships
        public Category Category { get; set; }
		public ICollection<OrderRequest> OrderRequests { get; set; }
	}
}

File: MyToolTrackerClient/src/types/tool.ts
export interface Tool {
  id: number;
  name: string;
  barcode: string;
  price: number;
  entryDate: Date; // date when the tool was added to the database
  model: string;
  manufacturer: string;
  categoryId: number; // references category of the tool
  orderRequestId: number; // references order request that the tool was ordered from
  toolStatusId: number; // references status of the tool
}
